os/arch/arm/src/amebasmart/amebasmart_serial.c

```diff
@@ -1103,6 +1097,9 @@ static void rtl8730e_up_txint(struct uart_dev_s *dev, bool enable)
 	struct rtl8730e_up_dev_s *priv = (struct rtl8730e_up_dev_s *)dev->priv;
 	DEBUGASSERT(priv);
 	priv->txint_enable = enable;
+#ifdef CONFIG_PM
+	bsp_pm_domain_control(BSP_UART_DRV, enable);
```

Comment:
Please check `bsp_pm_domain_control: Unable to resume bsp_drv_id: 4!` logs

pm_suspend and pm_resume should be called pair

Reply:
bsp_pm_domain_control(BSP_UART_DRV, enable); will be set to 1 and 0.
"enable" is set based on UART Tx request, where there is still data in the ring buffer, this txinit bit will be set to 1.
In uart_write(), uart_disabletxint() will disable UART Tx init due to there being no data in the buffer.

Thus, bsp_pm_domain_control(BSP_UART_DRV, enable); is called in pair.

Reply:
now, you can get `bsp_pm_domain_control: Unable to resume bsp_drv_id: 4!` error.

Could you please check again in your test?
Please check the errno number. if it is ERANGE, you are not matching pair of pm_suspend /pm_resume

Reply:
> bsp_pm_domain_control(BSP_UART_DRV, enable); will be set to 1 and 0. "enable" is set based on UART Tx request, where there is still data in the ring buffer, this txinit bit will be set to 1. In uart_write(), uart_disabletxint() will disable UART Tx init due to there being no data in the buffer.
> 
> Thus, bsp_pm_domain_control(BSP_UART_DRV, enable); is called in pair.

It is possible that uart_disabletxint() is invoked multiple times even the suspended domain has been resumed. Then, you might encounter the error mentioned by eunwoo.
Please check again the flow in os/drivers/serial/serialirq.c, serial.c. The BSP code is invoked by the TizenLite driver, and there is no guarantee that layer does that operation in pair, since both layer are maintained independently.

Reply:
you can use `uint16_t pm_suspendcount(int domain_id)`

Reply:
I have updated the PR, from my side I couldn't reproduce the issue.
I have changed the way of calling bsp_pm_domain_control().